app/js/HttpController.js

```diff
@@ -208,10 +208,18 @@ module.exports = HttpController = {
   zipProject(req, res, next) {
     const { project_id } = req.params
     logger.log({ project_id }, 'exporting project history as zip file')
-    ZipManager.exportProject(project_id, function (err, outputStream) {
-      pipeline(outputStream, res, err => {
-        if (err) logger.error({ project_id, err }, 'zip pipeline error')
-      })
+    ZipManager.exportProject(project_id, function (err, path) {
+      if (err) {
+        logger.error({ project_id, err }, 'zip preparation error')
+        next(err)
+      } else {
+        // send the file
+        res.download(path, `${project_id}-track-changes.zip`, function (err) {
+          if (err) {
+            logger.error({ project_id, err }, 'zip download error')
```

Comment:
I assume that a client disconnecting before having downloaded everything is going to trigger that error here, which we can't do much about. For that reason, I'd downgrade this log to info or warn